### PR TITLE
Setup GitHub actions

### LINKED ISSUE #616: Move Linux and MacOS CI jobs to GitHub Actions
Setup CI for GitHub Actions to replace Travis as travis.org will be decommissioned and will begin phasing in credit based system for Mac OS runs.

### PR DESCRIPTION
**Pull request by tijcolem**
Resolves #616 

https://travis-ci.org/ is shutting down and moving to travis-ci.com which uses a credit based system, at least for MacOS runs.  
This PR uses GitHub actions to run the Linux and MacOS jobs in replace of Travis. Appeyor CI jobs will remain the same and continue to run there unless something changes.  GitHub actions does support [Windows instances](https://github.com/actions/virtual-environments) so we could move 100% of our CI jobs there if desired.  

Once all tests are passing and PR is approved we can disable the Travis CI jobs. 

Checklist for GitHub Actions CI jobs: 

- [x] Run unit tests, integration, gem packaging and uploads for MacOS 10.15
- [x] Run unit tests, integration, gem packaging and uploads for Ubuntu 18.04 
- [x] Run integration tests (algo models) on docker compose stack. Uploads docker images. Ubuntu 18.04
- [x] Removed xcode 10.2 / osx 10.14 runs as these images are not available on GitHub actions.  
- [x] Github actions is configured to build Pull Requests and pushes to any branch (same as was on Travis).

### CODE REVIEW

File: bin/resources/local/rspec_command [hunk 1]
```diff
@@ -86,7 +86,7 @@ rspec_path = File.absolute_path(File.join(__FILE__, './../../../../gems/bin/rspe
 ::ENV['OS_SERVER_LOG_PATH'] = File.join(output_dir, 'logs', 'rails.log')
 ::ENV['RAILS_ENV'] = 'local-test'
 
-#If this is a travis build and it's BUILD_TYPE=test we want to omit the openstudio_algo_spec.rb tests
+#If this is a CI build (travis, github actions, etc..) and the BUILD_TYPE=test omit the openstudio_algo_spec.rb tests
```
Comment (nllong) [inline bin/resources/local/rspec_command:89]: 👍

File: ci/github-actions/cipher.rb [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+}
+puts
+puts
+
+alg  = ARGV.shift || ciphers.first
+pass = "secret password"
+salt = "8 octets"        # or nil
+text = "abcdefghijklmnopqrstuvwxyz"
+
+crypt_by_password(alg, pass, salt, text)
```
Comment (nllong) [inline ci/github-actions/cipher.rb:54]: extra line ending needed

File: ci/github-actions/cipher.rb [hunk 3]
```diff
@@ -0,0 +1,54 @@
+#!/usr/bin/env ruby
```
Comment (nllong) [inline ci/github-actions/cipher.rb:1]: looks like this is a new file to provide some basic salting/hashing of our passwords

File: ci/github-actions/export_build_linux.sh [hunk 4]
```diff
@@ -0,0 +1,20 @@
+# empty dir for export
```
Comment (nllong) [inline ci/github-actions/export_build_linux.sh:1]: sweet! finally building package from the CI!

**Review by nllong — APPROVED** [review]
It looks like you did a bunch of work to make this happen--good job!

Should we delete the `ci/travis` folder?

**Comment by tijcolem** [comment]
@nllong We can remove the  `ci/travis` folder. I left it in the event we needed to move back during the time of transition, but like all "temporary" things, it'll just live there indefinitely if we don't take care of it now.  =)  I'll make this change.

File: ci/github-actions/cipher.rb (hunk 3 above)
Comment (tijcolem) [reply ci/github-actions/cipher.rb:1]: We've had some issues with openssl in the past so @anyaelena added this as a sanity check during the gem builds.

**Comment by nllong** [comment]
> @nllong We can remove the `ci/travis` folder. I left it in the event we needed to move back during the time of transition, but like all "temporary" things, it'll just live there indefinitely if we don't take care of it now. =) I'll make this change.

Cool, thanks. Likewise, are we deprecating `AppVeyor` in favor of Windows GH actions.

**Comment by tijcolem** [comment]
AppVeyor remains the same. For now anyway.  I like the RDP debug features that AppVeyor provides (used it many times), so unless they make changes like Travis, I'd vote to keep it the same.  GitHub actions seems to provide the same debuging capabilities - https://github.com/stascorp/rdpwrap - but this doesn't seem as easy to enable/disable like AppVeyor.

**Comment by tijcolem** [comment]
Okay, all travis is now removed and disabled.

**Pull request merged**